spec/plugins/data_table/server_side.js

```diff
@@ -673,13 +673,79 @@ describe("DataTable Plugin", function() {
       table.render();
       jasmine.Ajax.requests.mostRecent().response(mockResponse.get());
+      history.pushState(null, null, "_SpecRunner.html");
     });
 
+    function clearFilters() {
+      var cg = table.configGenerator();
+      table.dataTable.find("thead th").each(function (index) {
+        var wrapper = $(".DataTables_sort_wrapper", this);
+        if (wrapper) {
+          var title = wrapper.text();
+          var col = cg.columnConfigByTitle.attributes[title];
+          if (col && col.filter) {
+            if (col.filter.type === "string") {
+              $(".toggle-filter-button", this).click();
+              $('.filterMenu input').val("").trigger("change");
+            }
+            else if (col.filter.type === "numeric") {
+              $(".toggle-filter-button", this).click();
+              $("input#first-filter").val("").trigger("change");
+            }
+            else if (col.filter.type === "list") {
+              $(".toggle-filter-button", this).click();
+              $('.filterMenu input').prop("checked", false).trigger("change");
+            }
+          }
+        }
+      });
+    };
+
+    function populateFilters() {
+      var cg = table.configGenerator();
+      table.dataTable.find("thead th").each(function (index) {
+        var wrapper = $(".DataTables_sort_wrapper", this);
+        if (wrapper) {
+          var title = wrapper.text();
+          var col = cg.columnConfigByTitle.attributes[title];
+          if (col && col.filter) {
+            if (col.filter.type === "string") {
+              $(".toggle-filter-button", this).click();
+              $('.filterMenu input').val("Scott").trigger("change");
+              $(".btn-filter").trigger("click");
+            }
+            else if (col.filter.type === "numeric") {
+              $(".toggle-filter-button", this).click();
+              $('select[data-filter-id=first-filter]').val("eq").trigger("change");
+              $('input#first-filter').val("0.5").trigger("change");
+              $(".btn-filter").trigger("click");
+            }
+            else if (col.filter.type === "list") {
+              $(".toggle-filter-button", this).click();
+              $(".filterMenu input[value=Basic]").prop("checked", true).trigger("change");
+              $(".btn-filter").trigger("click");
+            }
+          }
+        }
+      });
+    }
+
     function VerifyFilterAjax(filterObj) {
       var url = jasmine.Ajax.requests.mostRecent().url;
       var expectedFilterJson = encodeURIComponent(JSON.stringify(filterObj));
       expect(url).toMatch("ext_filter_json="+expectedFilterJson);
     }
+    function verifyUrlParams(filterObj) {
+      var uri = encodeURIComponent($.deparam(window.location.href.split("?")[1]).ext_filter_json)
+      var expectedFilterJson
+      if (filterObj.length>0) {
+        expectedFilterJson = encodeURIComponent(JSON.stringify(filterObj));
+      }
+      else {
+        expectedFilterJson = "";
+      }
```

Comment:
i think this would be a good use of a ternary
`var expectedFilterJson = (filterObj.length>0) ? encodeURIComponent(JSON.stringify(filterObj)) : expectedFilterJson = "";`